python/cuml/manifold/t_sne.pyx

```diff
@@ -509,6 +492,30 @@ class TSNE(Base,
             raise ValueError("Allowed methods are 'exact', 'barnes_hut' and "
                              "'fft'.")
 
+        cdef TSNEParams params
```

Comment:
Just a tiny nitpick- it might clean this method up even further if the TSNE class had a method like `_build_tsne_params` that could run this logic.